### PR TITLE
fix smarty flink function, if its a composer project

### PR DESCRIPTION
**Pull request by wrongspot**
### 1. Why is this change necessary?
If you use shopware composer project, the smarty flink function generates the wrong file include paths. Some of the JS Libraries linked from vendor paths. Usually vendor paths not accessable from outside, so if you try to use tinymce in shopware backend, you get forbidden errors.

the main header of the shopware backend
`themes/Backend/ExtJs/backend/base/header.tpl`
bildschirmfoto 2018-09-20 um 21 05 00

without this fix, you get this kind of js library includes in html source code
bildschirmfoto 2018-09-20 um 21 11 31

but in a composer project the libraries symlinked into that folder 
`/engine/Library`

so the real include path is
`/engine/Library/ExtJs/ext-all.js`

### 2. What does this change do, exactly?
It removes the vendor paths from the file paths.

### 3. Describe each step to reproduce the issue or behaviour.

- create an shopware composer project
- navigate to the shopware backend
- look at the HTML source code
- the javascript files (tiny_mce.js,ext-all.js,codemirror.js) will be include from vendor paths

### 6. Checklist

- [ ] I have written tests and verified that they fail without my change
- [ x] I have squashed any insignificant commits
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [x ] I have read the contribution requirements and fulfil them.

### CODE REVIEW

**Comment by MarcelSchmaeing** [comment]
Hi,

thx for the PR. You need to sign the license/cla otherwise we cannot accept your PR.

Greetings,

Marcel

**Comment by wrongspot** [comment]
should signed now

Greetings Björn

File: engine/Library/Enlight/Template/Plugins/function.flink.php [hunk 1]
```diff
@@ -59,6 +59,11 @@ function smarty_function_flink($params, $template)
             }
         }
 
+        //Cleanup vendor paths if is composer project
+        if (strpos($file, 'vendor/shopware/shopware') !== false) {
```
Comment (hlohaus) [inline engine/Library/Enlight/Template/Plugins/function.flink.php]: if strpos but in second line it is substring with docPath?

Comment (wrongspot) [reply engine/Library/Enlight/Template/Plugins/function.flink.php]: yes, you are right. thats not clear enough. I rewrite that in another commit.

**Comment by wrongspot** [comment]
Is that really an error of my commit?

**Comment by nlubisch** [comment]
No, its an error in the main branch. You can ignore it - shopware will fix it.

**Comment by wrongspot** [comment]
again the same error, mhhh

**Comment by nlubisch** [comment]
You have to update your 5.5 branch, than rebase your feature-branch onto the updated 5.5 branch.

**Comment by bragento** [comment]
You can close this PR, is fixed with this commit https://github.com/shopware/shopware/commit/5def646105d36305da5159ac3935fc1f808f95e2

**Comment by wrongspot** [comment]
Sorry, my mistake. This bug is allways there.

We fix it in currently in app/functions.sh with rsync instaed of symlinking.

```
    #ln -s ../../vendor/shopware/shopware/engine/Library/CodeMirror engine/Library/CodeMirror
    #ln -s ../../vendor/shopware/shopware/engine/Library/ExtJs engine/Library/ExtJs
    #ln -s ../../vendor/shopware/shopware/engine/Library/TinyMce engine/Library/TinyMce
    rsync -az --delete vendor/shopware/shopware/engine/Library/CodeMirror/ engine/Library/CodeMirror/
    rsync -az --delete vendor/shopware/shopware/engine/Library/ExtJs/ engine/Library/ExtJs/
    rsync -az --delete vendor/shopware/shopware/engine/Library/TinyMce/ engine/Library/TinyMce/
```

**Comment by alphanyx** [comment]
We notices the same Error in Shopware 5.5.4.
So the "bug" wasn't fixed

**Comment by wrongspot** [comment]
So the issue is fixed, with this PR https://github.com/shopware/composer-project/pull/28/commits/f2a20dc48ed151d73463e1fd7d50b668861f2a30

**Pull request closed without merging**